src/Sulu/Bundle/PageBundle/Controller/IconController.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\PageBundle\Controller;
+
+use FOS\RestBundle\View\ViewHandlerInterface;
+use HandcraftedInTheAlps\RestRoutingBundle\Routing\ClassResourceInterface;
+use Sulu\Component\Rest\AbstractRestController;
+use Sulu\Component\Rest\ListBuilder\CollectionRepresentation;
+use Symfony\Component\Finder\Finder;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+class IconController extends AbstractRestController implements ClassResourceInterface
+{
+    /**
+     * @var array<array<array{path: string, selection_json: string}>>
+     */
+    private array $iconSets;
+
+    /**
+     * @param array<array<array{path: string, selection_json: string}>> $iconSets
+     */
+    public function __construct(
+        ViewHandlerInterface $viewHandler,
+        array $iconSets,
+    ) {
+        parent::__construct($viewHandler);
+        $this->iconSets = $iconSets;
+    }
+
+    /**
+     * Returns icons.
+     *
+     * @return Response
+     */
+    public function cgetAction(Request $request)
+    {
+        $iconSet = (array) $this->iconSets[$request->query->get('icon_set')];
+        $search = $request->query->get('search');
+        $provider = $iconSet['provider'];
+        $icons = [];
+
+        // Get icons depending on the provider type.
+        switch ($provider) {
+            case 'svg':
+                $icons = $this->getIconsAsSVGs($iconSet);
+                break;
+            case 'icomoon':
+                $icons = $this->getIconsAsIcomoon($iconSet);
+                break;
+        }
```

Comment:
maybe we should throw an exception here if we dont know how to handle the provider?